File: 3.数组中重复的数字/3.数组中重复的数字.cpp
class Solution {
public:
    // Parameters:
    //        numbers:     an array of integers
    //        length:      the length of array numbers
    //        duplication: (Output) the duplicated number in the array number
    // Return value:       true if the input is valid, and there are some duplications in the array number
    //                     otherwise false
    bool duplicate(int numbers[], int length, int* duplication) {
        for(int i=0;i<length;i++){
            int val = numbers[i];
            if(i!=val){//避免第一次相等的判断
                if (numbers[numbers[i]]==val){
                *duplication = val;
                return true;
                }
            }
            numbers[i] = numbers[val];
            numbers[val] = val;
        }
        *duplication = -1;
        return false;
    }
};

File: 3.数组中重复的数字/3.数组中重复的数字.py
# -*- coding:utf-8 -*-
class Solution:
    # 这里要特别注意~找到任意重复的一个值并赋值到duplication[0]
    # 函数返回True/False
    def duplicate(self, numbers, duplication):
        # write code here
        for i, val in enumerate(numbers):
            if i!=val:
                if numbers[numbers[i]]==val:
                    duplication[0] = val
                    return True
                numbers[i] = numbers[val]
                numbers[val] = val
        return False
